### ISSUE TITLE
Are all tasks required for WCAG conformance

### ISSUE THREAD

**Issue by sarahpulis** [w3c/wai-arrm#69]
I was wondering what the rationale is for tasks in ARRM being associated with its success criterion.

For example, one of the tasks ANM-024 states:

> Volume controls for page-level multimedia files are visually located at the top of the page.

The success criterion for this task is1.4.2 Audio Control.

The Success Criterion 1.4.2 Audio Control states:

> If any audio on a web page plays automatically for more than 3 seconds, either a [mechanism](https://www.w3.org/WAI/WCAG22/Understanding/audio-control.html#dfn-mechanism) is available to [pause](https://www.w3.org/WAI/WCAG22/Understanding/audio-control.html#dfn-paused) or stop the audio, or a mechanism is available to control audio volume independently from the overall system volume level.

My understanding of this criterion is that if the volume controls are **not** located at the top of the page, then it would not fail 1.4.2. The criterion doesn't require the mechanism to be at the top of the page, but only that it is available. 

It is definitely best practice, and is noted in Understanding 1.4.2 Audio Controls in the example:

> An audio file begins playing automatically when a page is opened. However, the audio can be stopped by the user by selecting a "silent" link at the top of the page.

There seems to be multiple instances where there are WCAG requirements (tasks that would fail a success criterion if not implemented) and best practice requirements (tasks that would not fail a success criterion if not implemented) which are all listed against a success criterion in the same way. This may be misleading if people are assuming all tasks need to be implemented to meet the associated success criterion, given they are all presented in the same way within the [Tasks involved in Accessibility](https://www.w3.org/WAI/planning/arrm/tasks/).

**Comment by mgifford** [comment]
Thanks for this @sarahpulis & @msales78 - both of your comments are great.

Sarah, the concept behind the task ID's was to break apart the SC into smaller components that we felt could be better assigned to a specific role. Some of the SC are broad enough that they touch on a lot skills from a digital team. We wanted to not look first from the user, but from the role. 

That said, good observation with [ANM-024 & 1.4.2 (A](https://www.w3.org/WAI/planning/arrm/tasks/)). There were a small team of us that worked on this (some for close to a decade now). But the ARRM process is certainly less vetted than WCAG. We are opening up so that we can help look at WCAG application in a new way. 

On that issue, thinking just about User Experience (UX) Design, is there a way that you would express how to place the volume controls for page-level multimedia files? 

Marcelo, I do like the idea of "use cases" I think this is an example of a new issue that we should spin up. Flipping it about, it is interesting to see how this Brazilian standard delt with 1.4.2 (A):

Audio Control - 1.4.2 (A) - ABNT 5.14.7  -  UX & DEV
- There is no audio that plays automatically and lasts longer than 3s; or
- There is a mechanism to pause, stop, mute, or adjust your volume without affecting the overall volume of the system. Requirement 

It is interesting to see what alternative ways folks can express similar ideas.  Also, interesting that this standard is split down by UI/UX. 

There are no right ways to divide up content like this, but there are always better ways. I don't know what the best way is, but this is a great place for that, if we can bring the right people into the discussion.

**Comment by anna-yeddi** [comment]
Another option to clarify the tasks may be adding `, if present`, for instance, in the case ANM-024, could be:
> Volume controls for page-level multimedia files, if present, are visually located at the top of the page.

Yet, I do agree with the comments aboce that providing the either-or options would also serve this well. For instance, when a product is tested against WCAG, it would pass, if there are no audio that auto-plays _or_ if the mechanism is provided. And this split in tasks could be a way to highlight both options.

**Comment by dboudreau** [comment]
Hello everyone. Interesting conversation @sarahpulis, thanks for bringing that up. It's the perfect opportunity for me to clarify the intent of this resource.

First, the ARRM resource is not meant to replace WCAG, nor should any of the interpretations of any of the tasks be perceived as anything normative. What these tasks are is just a representation of some of the things that the success criteria suggest - or recommendable best practices related to the SC. They are in any way, shape, or form, meant to replace what the normative and informative sections of WCAG provide in the TR or the Understanding documents.

However, that being said, I can absolutely understand how some people might interpret these tasks and their presentation within ARRM as "gospel" and feel uncomfortable when it moves away from what might be a different, maybe even more agreed upon interpretation. What you bring up with 1.4.2 is a good example of that.

The solution is not necessarily to rewrite any task that does not fall completely within the boundaries of what is required by WCAG, but rather to clearly explain that these tasks are just a way to get closer to WCAG conformance. As they stand out, every one of those tasks brings a user interface closer to accessibility. And because we needed to connect everything to a success criterion, some associations ended up being more natural than others... but it's not always a perfect match, as is the case with the example that you brought up with audio control. 

Of course, we could go through all 300+ tasks and label them as either A) a best practice related to a success criterion or B) an actual aspect of a success criterion, but this work goes far beyond the scope of this current project and was never even meant to be in scope from the beginning. But it is something that could be taken as a future initiative within this community group, because every task should be revisited as often as needed to keep them fresh and relevant. 

For this first iteration, that is not what ARRM is intended for. This resource is not, nor was it ever meant, as a replacement for WCAG or any of the official W3C documents. What this resource is intended for is bringing digital professionals closer to accessibility by giving them a means to better understand what the success criteria are about and what they might mean for their jobs. And in that way, the tasks as they are written right now do this pretty well.

Of course, everything can always be improved, and we have every intention of doing so over the next couple years, especially as qualified volunteers like yourself join us in this effort. This is why it made a lot of sense to turn ARRM into a community group, so that we can get the feedback from everybody and have all those contributions in order to make the resource better, more thorough, and more aligned with the language and expectations of those professionals. As our numbers grow, we could tackle such things and iron out discrepancies.

Labelling each task as they are written as best practices or requirements is an option, but it would introduce its own set of challenges, and that particular Pandora's box is not something that we should really consider at this time. First of all, that would mean months of conversations to get to an agreement as to what every success criterion is actually intended for, and where the bar is set in terms of expectations. 

This would be counterproductive, as decades of conversations about personal interpretations dictate. And while we'd be busy doing that in our ivory tower, professionals out there would not be hearing the message that they need to hear, which is that there are things they can do to make their digital content more inclusive. Again, these tasks, as they are currently presented, already do that quite well despite some irregularities that still bring professionals closer to accessibility.

We need to remember that no one out there is interested in the fine details of what we, as W3C participants, actually mean or intend when we come up with a success criterion. What they care about is clarity as to what they're supposed to do to make their content more accessible, so they can move the needle toward more inclusive interfaces and break down barriers for their users living with disabilities. As an educator, I know that sometimes, cutting a corner or two is the best thing we can do to educate people, and this is what this resource allows for every now and then to smoothen that learning curve.

The tasks, as they are written and no matter how imperfect they might be right now, all contribute to moving that needle a little further. The ARRM resource is not meant to replace WCAG, and therefore, should not be regarded as a replacement for what is already captured elsewhere. It is simply meant, and was always meant, as an intermediary step to understanding the Understanding documents, so that ultimately people can better understand the success criteria.

The solution to me, because I recognize the validity of your comment, is to address that perspective head on with some disclaimer that clearly states "yes, there can be the discrepancies between what ARRM suggests and what WCAG says, and in those situations, WCAG should always prevail". 

What I believe people are looking for is guidance and clarity to get closer to conformance because they can't quite wrap their heads around the complexity of the Understanding documents. Thy need something simpler to understand and the tasks as presented are a great way for them to do that, while broadening and scaffolding their perspective towards more inclusive, digital content.

If you are interested in suggesting changes to tasks or to the structure, by all means, please do so - join our calls if at all possible given your timezone - and let's make this resource even better. We could certainly use the bright minds at Intopia. :)

**Comment by dboudreau** [comment]
@sarahpulis gentle nudge to see what do you think about my comment above? We'd like to close this issue if you're on board with our framing of the resource.

**Comment by shawna-slh** [comment]
Thank you for raising this issue, @sarahpulis and for the reply @dboudreau 

I think this is a significant issue that should be addressed as a top priority.

It is common for implementers and evaluators to debate what is actually required by WCAG versus what is best practice. This GitHub issue highlights that ARRM could potentially further complicate this challenge if anyone thinks the ARRM Tasks are required to meet WCAG.

Ideas for addressing it:
- Wherever Tasks are listed, provide clear, strong statement that the Tasks are suggestions and not necessarily required to meet WCAG or sufficient to meet WCAG.
  - Put that in an expanded [Tasks page Introduction](https://www.w3.org/WAI/planning/arrm/tasks/#introduction).
  - In the role pages under 'Tasks to get started', add something brief and link to there.
- Assess the Tasks that are not required to meet WCAG. Consider removing these or clearly indicating that they are not required. How to do this will take some thought. I'm happy to help consider options.

**Comment by dboudreau** [comment]
As a way to follow up on this issue, we're going to revisit the content types one by one, review every task therein and assign each one a label:

* Required = needed to meet the WCAG success criterion as written.
* Conditionally required = required when that situation exists.
* Best practice = helpful, but not required for conformance.

Once we all agree on the interpretation for each task, we will add this info to the resource so this issue can feel satisfactory.

- [ ] **IMG-001. / SC 1.1.1: Required** (Informative images need a text alternative)
- [ ] **IMG-002. / SC 1.1.1: Required** (A meaningful equivalent is required. “Clear” is guidance)
- [ ] **IMG-003. / SC 1.1.1: Required** (Decorative images must be ignorable by assistive tech)
- [ ] **IMG-004. / SC 1.1.1: Required** (Redundant images should not repeat nearby text)
- [ ] **IMG-005. / SC 1.1.1: Best practice** (Separate links can still pass if both are named)
- [ ] **IMG-006. / SC 1.1.1: Required** (Images of text must have that text as an alternative)
- [ ] **IMG-007. / SC 1.1.1: Best practice** (CSS images can pass if text alternatives exist)
- [ ] **IMG-008. / SC 1.1.1: Required** (Complex images need an equivalent description of purpose)
- [ ] **IMG-009. / SC 1.1.1: Best practice** (Alt is one method. Other equivalent methods exist)
- [ ] **IMG-010. / SC 1.1.1: Conditionally required** (Needed when short alt cannot convey full meaning)
- [ ] **IMG-011. / SC 1.1.1: Best practice** (WCAG needs access to the info, not a “mechanism.”)
- [ ] **IMG-012. / SC 1.1.1:  Best practice** (longdesc is optional. Any equivalent method works)
- [ ] **IMG-013. / SC 1.1.1: Required** (Functional images need purpose, not visual description)
- [ ] **IMG-014. / SC 1.1.1: Best practice** (Avoiding “link/button” in alt improves UX, not conformance)
- [ ] **IMG-015. / SC 1.1.1: Conditionally required** (If the image changes, the text alternative must match)
- [ ] **IMG-016. / SC 1.1.1: Required** (CAPTCHA needs an alternative modality. WCAG requires this)
- [ ] **IMG-017. / SC 1.1.1: Best practice** (Classification helps process, but WCAG tests the outcome)
- [ ] **IMG-018. / SC 1.1.1: Best practice** (This is mainly 1.4.1 Use of Color, not 1.1.1)
- [ ] **IMG-019. / SC 1.4.5: Required** (Don’t use images of text when real text works)
- [ ] **IMG-020. / SC 1.4.5: Best practice** (HTML/CSS is a technique, not a requirement. Outcome matters)
- [ ] **IMG-021. / SC 1.4.5: Best practice** (Too strict and slightly off. 1.4.5 allows “essential.”)
- [ ] **IMG-022. / SC 1.4.9: Required** (AAA only - images of text must be decorative only. Logos are exempt)

* All AI generated to launch the ocnversation.

**Comment by mgifford** [comment]
I think this is accurate in the PR https://deploy-preview-117--wai-arrm.netlify.app/planning/arrm/tasks/

@dboudreau if you can confirm.

**Comment by JennChadwick** [comment]
@sarahpulis Thank you for your email to me on 05-04-2026 (May 5th, 2026). 

In it, you said: "Thank you for the update about ARRM. I am glad my comment has led to something that is hopefully practical and useful for all."

Noting this in comments.  Thanks.

**Comment by dboudreau** [comment]
@JennChadwick Thank you for getting the confirmation from @sarahpulis.

Based on this, it is now time to close this issue and move forward. As a next step - and to address @shawna-slh's concern - we will be adding the following paragraph at the top of the Tasks page, for clarity about authoritative/normative sources:

**Important:** ARRM is a practical implementation resource, not a normative interpretation of WCAG. The tasks it contains are intended to help teams move closer to WCAG conformance by translating success criteria into actionable activities and highlighting recommended best practices. They do not replace the normative WCAG specification or the informative Understanding WCAG documents, and they should not be interpreted as definitive or exhaustive requirements. While each task is associated with a relevant success criterion, some associations are stronger than others. In some cases, a task reflects a recommended practice that supports accessibility rather than a direct requirement of the referenced success criterion. When questions arise, WCAG and its supporting documentation remain the authoritative source.
